### ISSUE TITLE
[Bug Report]: When using @pandacss/dev, the project works on development but breaks on production

### ISSUE THREAD

**Issue by tcdw** [farm-fe/farm#1644]
### Steps to reproduce

1. Clone the reproduction repo and enter the directory
2. Run`pnpm install`
3. Run `pnpm panda codegen`
4. Run `pnpm run dev`. (As you can see, it seems working fine)
5. Run `pnpm run build` then preview that with `pnpm run preview`

### Reproduce link

https://github.com/tcdw/farm-pandacss-test

### What is actually happening?

The page didn't show anything, which is not expected compared to development, and the console has errors like this:

![prod-error](https://github.com/user-attachments/assets/03a390c3-08af-41b6-b658-dfe1e0bfc196)

### System Info

```Shell
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Pro
    Memory: 181.30 MB / 16.00 GB
    Shell: 5.2.26 - /opt/homebrew/bin/bash
  Binaries:
    Node: 20.12.2 - /Applications/ServBay/bin/node
    Yarn: 1.22.21 - ~/Library/pnpm/yarn
    npm: 10.5.0 - /Applications/ServBay/bin/npm
    pnpm: 9.1.2 - ~/Library/pnpm/pnpm
    bun: 1.1.6 - ~/.bun/bin/bun
    Watchman: 2024.07.08.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 126.0.6478.183
    Edge: 126.0.2592.102
    Safari: 17.5
  npmPackages:
    @farmfe/core: ^1.3.0 => 1.3.5
```

**Comment by Nirvana-Jie** [comment]
It seems that the demo you provided cannot be reproduced. Can you provide more information?
<img width="392" alt="image" src="https://github.com/user-attachments/assets/2d79ef75-503b-41d4-b05f-fc054440acab">
<img width="898" alt="image" src="https://github.com/user-attachments/assets/14b871e8-2f92-4c3d-a505-f3b73b336995">

**Comment by tcdw** [comment]
> It seems that the demo you provided cannot be reproduced. Can you provide more information?

I switched to another Windows computer, and the problem still occurs.

Here is the information of that Windows computer:

```text
  System:
    OS: Windows 11 10.0.22631
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700
    Memory: 47.17 GB / 79.70 GB
  Binaries:
    Node: 20.15.1 - ~\AppData\Local\nvs\default\node.EXE
    npm: 10.7.0 - ~\AppData\Local\nvs\default\npm.CMD
    pnpm: 9.5.0 - ~\AppData\Local\pnpm\pnpm.CMD
    bun: 1.1.6 - ~\.bun\bin\bun.EXE
  Browsers:
    Edge: Chromium (125.0.2535.92)
    Internet Explorer: 11.0.22621.3527
  npmPackages:
    @farmfe/core: ^1.3.0 => 1.3.5
```

And after recheck my issue, I found I lacked a step for generate codes with Panda CSS. That's my bad, and the original issue has been edited.

Here is how I expect it looks like in production:

![error2](https://github.com/user-attachments/assets/ba0352b7-2a1d-4f91-8f8b-e6c596bf60f4)

**Comment by tcdw** [comment]
Well, I found a workwround for myself, and at least the project runs on production:

```typescript
export default defineConfig({
    compilation: {
        minify: false,
    },
})
```

**Comment by wre232114** [comment]
you can upgrade your @farmfe/core to lastest, we fixed a minify bug in 1.3.6

**Comment by tcdw** [comment]
> you can upgrade your @farmfe/core to lastest, we fixed a minify bug in 1.3.6

Thanks for your reply, I am really appreciated. Unfortunely, after updating `@farmfe/core` to `^1.3.7` in my reproduction repo, it still didn't seem working.

It still complaints `Uncaught TypeError: m is not a function` as seen in my first screenshot.

To ensure my environment is configured correctly, I also configured GitHub Actions, and you can see the deployed result [here](https://tcdw.github.io/farm-pandacss-test/).

**Comment by wre232114** [comment]
try set minify: { moduleDecls: false } for production to see if it works. I'll look into your project

**Comment by tcdw** [comment]
> try set minify: { moduleDecls: false } for production to see if it works. I'll look into your project

Wow, it saved my life! The project now works in production with JS file mostly minified. 😇

**Comment by chneau** [comment]
To add more context, I had the same issue with some Redux exported method, `compilation: { minify: { moduleDecls: false } }` seems to fix it.

**Comment by wre232114** [comment]
this issue is fixed in 1.3.8

**Comment by tcdw** [comment]
Thank you!
